File: src/main/java/org/onlineoptimisation/problem/benchmark/SphereFunction.java
package org.onlineoptimisation.problem.benchmark;

import org.armadillojava.Arma;
import org.armadillojava.Col;
import org.onlineoptimisation.problem.OptimisationProblem;

public class SphereFunction extends BenchmarkProblem {

  public SphereFunction(int numberOfDimensions) {
    super(numberOfDimensions);
  }

  @Override
  public double getObjectiveValueImplementation(Col parameter) {
    return Arma.accu(Arma.square(parameter));
  }

}

File: src/optimisationProblem/benchmark/bbob2013/sphereFunction.cpp
#include <hop_bits/optimisationProblem/benchmark/bbob2013/sphereFunction.hpp>

#include <cmath>

#include <cereal/archives/json.hpp>
#include <cereal/types/polymorphic.hpp>

namespace hop {
  double SphereFunction::getObjectiveValueImplementation(const arma::Col<double>& parameter) const {
    return std::pow(arma::norm(parameter - translation_), 2);
  }

  std::string SphereFunction::to_string() const {
    return "SphereFunction";
  }
}

CEREAL_REGISTER_TYPE(hop::SphereFunction)
